agent/app/run.go

```diff
@@ -43,7 +49,14 @@ func Run(arguments []string) int {
 		// timeout of 30s. This is so that we can catch any http timeout and log the
 		// issue within agent logs.
 		// see https://docs.docker.com/engine/reference/builder/#healthcheck
-		return runHealthcheck("http://localhost:51678/v1/metadata", time.Second*25)
+		// if ECS_AGENT_HEALTHCHECK_HOST env var is set, it will override
+		// the healthcheck's localhost ip address inside the ecs-agent container
+		localhost := "localhost"
+		if localhostOverride, ok := os.LookupEnv(ECS_AGENT_HEALTHCHECK_HOST_ENV_VAR); ok {
```

Comment:
sorry for the back and forth on this, but I dont think this is correct.....`ok` will be true if the env var was set and empty, is that what we want? for users to be able to set the host to a blank string?

see https://play.golang.org/p/MLN0QyG_ray

cc @angelcar

Reply:
Ah, you're right cam. I had that backwards. Sorry @fierlion, I gave you bad advice with `LookupEnv` :(

Reply:
reverted.  see latest update.
nice catch Cam.